From 087318f56997705595f5ad51e4e67a8660259aae Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 06 Oct 2020 19:47:41 +0300
Subject: [PATCH] [SW-12754] gaudi: security: keep 2 more regs open

We keep mmPREBOOT_PCIE_EN and mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS
registers open meanwhile.
This is needed for smooth integration until FW gains ability to operate
those registers in a stable way (this includes reset sequence and WDOG).

Change-Id: I8817941d498cc47602d68c8b78792630a86696bb
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8ce32a1..d1dfed3 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -379,9 +379,16 @@
 	int i;
 	u32 mask, prot_base, base = mmPSOC_GLOBAL_CONF_BASE & ~HL_CFG_BASE;
 
+	/*
+	 * TODO: the register below is open temporarily to allow
+	 * LKD master to work correctly and condition the PCIe link init
+	 * properly.
+	 * mmPREBOOT_PCIE_EN (mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_1).
+	 */
 	prot_base = base;	/* offsets: 0 - 0x80 */
 	mask = ~(PROT_BIT(mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_0) &
-		PROT_BIT(mmPSOC_GLOBAL_CONF_BTM_FSM));
+		PROT_BIT(mmPSOC_GLOBAL_CONF_BTM_FSM) &
+		PROT_BIT(mmPREBOOT_PCIE_EN));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0x100 */
@@ -406,10 +413,20 @@
 		PROT_BIT(mmPSOC_GLOBAL_CONF_TRACE_ARUSER));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x480 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 2);
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x400 */
+	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE);
 
-	prot_base += SNGL_REG_PROT_SIZE * 2;	/* offsets: 0x480 - 0x500 */
+	/*
+	 * TODO: the register below is open temporarily to allow
+	 * LKD master to work correctly and condition the PCIe link init
+	 * properly.
+	 * mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS
+	 */
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x400 - 0x480 */
+	mask = ~PROT_BIT(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
+
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x480 - 0x500 */
 	mask = ~PROT_BIT(mmPSOC_GLOBAL_CONF_SW_ALL_RST);
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
